fdbclient/BackupContainer.actor.cpp

```diff
@@ -348,8 +348,23 @@ class BackupContainerFileSystem : public IBackupContainer {
 		return writeFile(snapshotFolderString(snapshotBeginVersion) + format("/%d/", snapshotFileCount / (BUGGIFY ? 1 : 5000)) + fileName);
 	}
 
+	// Find what should be the filename of a path by finding whatever is after the last forward or backward slash, or failing to find those, the whole string.
+	static std::string fileNameOnly(std::string path) {
+		// Find the last forward slash position, defaulting to 0 if not found
+		int pos = path.find_last_of('/');
+		if(pos == std::string::npos) {
+			pos = 0;
+		}
+		// Find the last backward slash position after pos, and update pos if found
+		int b = path.find_last_of('\\', pos);
+		if(b != std::string::npos) {
+			pos = b;
+		}
```

Comment:
This feels ever so slightly sketchy that you're matching / as a path separator on windows, and \ as a path separator on linux.  Given that this is the file container, I think ideally you'd move this into platform.cpp as relativeBasename() and use CANONICAL_PATH_SEPARATOR from there?

Reply:
IRL Steve says this is because Windows talking to S3 still needs to use forward slash, so this gets weirdly complicated, but as backup will never generate filenames with / or \ in them, it's fine.